co-authors-plus.php

```diff
@@ -713,12 +717,12 @@ function posts_where_filter( $where, $query ) {
 				}
 				$terms_implode = rtrim( $terms_implode, ' OR' );
 
-				$id = is_author() ? get_queried_object_id() : '\d+';
+				$id = is_author() && $query->is_main_query() ? get_queried_object_id() : '\d+';
 
 				//If we have an ID but it's not a "real" ID that means that this isn't the first time the filter has fired and the object_id has already been replaced by a previous run of this filter. We therefore need to replace the 0
 				// This happens when wp_query::get_posts() is run multiple times.
-				if ( false === get_user_by( 'id', $id ) ){
-					$id = 0;
+				if ( $id !== '\d+' && false === get_user_by( 'id', $id ) ){
```

Comment:
The check for `$id !== '\d+'` is also not necessary.  I think we can just have it assigned to `'\d+'` if `get_user_by( 'id', $id )` returns `false` since `$id` will not equal `'\d+'` if `$id` ends up equating to `get_queried_object_id()` which will not be `\d+`.